yargs.js

```diff
@@ -287,6 +290,25 @@ function Yargs (processArgs, cwd, parentRequire) {
     return options.demanded
   }
 
+  var recommendCommands
+  self.recommendCommands = function (value) {
+    if (value === undefined || !!value === true) {
+      recommendCommands = true
+    } else {
+      recommendCommands = false
+    }
+    return self
+  }
+
+  self.recommendCommand = function (cmd, handlerKeys) {
+    if (recommendCommands) {
+      const similarCommand = require('didyoumean')(cmd, handlerKeys)
+      if (similarCommand) {
+        console.log(__('Did you mean %s?', similarCommand))
```

Comment:
@nexdrew I agree that to make `self.recommendCommand` more reusable inside yargs, it makes sense to just return the determined value. However, if `.recommendCommands` is enabled, I'd still log it to the console since I think the idea behind this method is to assist the user. I can't really think of an advantage when `.recommendCommands()` just returns you a recommended command and you still have to log it, since that's what I think will happen anyway. What do you think?